geode-core/src/main/java/org/apache/geode/internal/statistics/StatMonitorHandler.java

```diff
@@ -50,36 +50,26 @@ public StatMonitorHandler() {
 
   /** Adds a monitor which will be notified of samples */
   public boolean addMonitor(StatisticsMonitor monitor) {
-    synchronized (this) {
-      boolean added = false;
-      List<StatisticsMonitor> oldMonitors = this.monitors;
-      if (!oldMonitors.contains(monitor)) {
-        List<StatisticsMonitor> newMonitors = new ArrayList<StatisticsMonitor>(oldMonitors);
-        added = newMonitors.add(monitor);
-        this.monitors = Collections.unmodifiableList(newMonitors);
-      }
-      if (!this.monitors.isEmpty()) {
-        startNotifier_IfEnabledAndNotRunning();
-      }
-      return added;
+    boolean added = false;
```

Comment:
I don't think it's safe to remove the synchronization here. You're calling contains followed by add followed by isEmpty. Also, startNotifier_... is dependent on the state of this.monitors. this.monitors could be getting changed by other threads at any point in here.